aiormq/connection.py

```diff
@@ -731,6 +758,32 @@ async def channel(
 
         return channel
 
+    async def update_secret(
+        self, new_secret: str, *,
+        reason: str = '', timeout: TimeoutType = None,
+    ) -> spec.Connection.UpdateSecretOk:
+        channel_frame = ChannelFrame(
+            channel_number=0,
+            frames=[
+                spec.Connection.UpdateSecret(
+                    new_secret=new_secret, reason=reason
+                )
+            ]
+        )
+
+        async with self.__update_secret_lock:
+            self.__update_secret_future = self.loop.create_future()
+            await self.write_queue.put(channel_frame)
+            try:
+                response: spec.Connection.UpdateSecretOk = (
+                    await asyncio.wait_for(
+                        self.__update_secret_future, timeout=timeout
+                    )
+                )
+            finally:
```

Comment:
Don't we want to update url to store new secret there to achieve secret rotation? 
We will be able to close and open connection with new secret (robust in aio-pika e.g.)

Reply:
This class doesn’t store the password after connection for security reasons.